### PR TITLE
feat: table of contents on documentation

### LINKED ISSUE #3867: Add a table of contents to documentation pages
Hello everyone, 

I have been working with Electron for a ling time now. I often refer to documentation and I noticed I spent a lot of time looking for a specific item before finding it. For example the [BrowserWindow](https://electronjs.org/docs/api/browser-window) is very long.

I cloned the repo and noticed you just render the markdown. 

I know I am reopening an issue but thy can't you analyse it in order to create a table of content? You could put it in a collapsible panel in order not to take more place.

I don't know how to code on your website but **I made a CodePen with a preview** of what it could look like with your style :
[https://codepen.io/TheRolf/full/wvKWOYK](https://codepen.io/TheRolf/full/wvKWOYK)
The demo is interactive, you can click the "Table of Contents" title to expand or collapse the table of content.

The script algorithm behind however is not very complicated. I could help. Just parse the markdown to create a hierarchy with sections, sub sections and platform/deprecated labels.

Thanks for reading!

### PR DESCRIPTION
**Pull request by erickzhao**
Closes #3867

Adds a new table of contents on the right side of the docs page based on prior work done in #4475 and does a few other things:
* Moves the doc actions (edit, translate) from the footer to the header.
* Disables https://github.com/electron/electronjs.org/pull/1053 (The duplicated DOM used by this selector doesn't play well with grabbing all of the headings in a page)

### CODE REVIEW

File: data/locale.yml [hunk 1]
```diff
@@ -39,11 +39,10 @@ docs:
   main_proc_modules: Main Process Modules
   renderer_proc_modules: Renderer Process Modules
   both_proc_modules: Modules for Both Process Types
-  footer:
-    propose_change: Improve this doc
-    version_history: Version history
-    translate: Translate this doc
-    view_all_docs: All Docs on One Page
+  propose_change: Improve this doc
```
Comment (erickzhao) [inline data/locale.yml:42]: These are no longer in the footer, so I've changed the yaml

File: public/styles/_base.scss [hunk 2]
```diff
@@ -144,4 +142,5 @@ img[data-src] {
 
 .PRIMER-REMOVE-ME {
   @import "@primer/css/index.scss";
+  @import 'tocbot/src/scss/_tocbot-core.scss';
```
Comment (erickzhao) [inline public/styles/_base.scss:145]: Importing the default tocbot styles

File: public/styles/_docs.scss [hunk 3]
```diff
@@ -1,209 +1,203 @@
...
+    font-size: $body-font-size * 0.95;
+    line-height: 1.8;
+    font-weight: 200;
+    -webkit-font-smoothing: initial;
+    background-color: $main-bg-color;
+    border: 1px solid $border-color;
+    border-radius: $border-radius;
+  }
 
-.docs-breadcrumbs {
```
Comment (erickzhao) [inline public/styles/_docs.scss:22]: The removed classes here were dead css

File: public/styles/_docs.scss [hunk 4]
```diff
@@ -1,209 +1,203 @@
 
 // Docs ------------------------------
 
-.docs-page {
-  a { color: #3a585f; }
-}
+.PRIMER-REMOVE-ME {
```
Comment (erickzhao) [inline public/styles/_docs.scss:4]: Wrapped the entire file with `.PRIMER-REMOVE-ME` so the diff shows a lot of whitespace changes

**Review by Kilian — APPROVED** [review]

**Comment by Kilian** [comment]
Nice addition! Works well, code looks good.

We could consider having a fixed "Table of contents" button on mobile viewports (where there is no place for the right sidebar) that opens and collapses the table of contents overlaid on the page (I do this on https://polypane.app/docs) but that can be a different ticket.

**Pull request merged**